Makefile

```
@@ -1,3 +1,12 @@
 .PHONY: update-licenseheaders
 update-licenseheaders:  ## add or update license headers in all python files
 	licenseheaders -y 2023 --owner "The Superpower Institute Ltd" --projname "OpenMethane" --tmpl .copyright.tmpl --ext .py -x "venv/*"
+
+.PHONY: virtual-environment
+virtual-environment:  ## update virtual environment, create a new one if it doesn't already exist
+	poetry lock
```

I prefer not automatically forcing all dependencies to the latest version (this still checks that the lock file is consistent with the pyproject.toml file)

```suggestion
	poetry lock --no-update
```